### PR TITLE
Validate .tld for new feed URLs

### PR DESCRIPTION
**Pull request by blowery**
Ensure that new feed URLs have a valid .tld before allowing the user to follow the feed.

dfbc6e9c-8eb5-11e5-8bec-c09eb793c6cd

Fixes Automattic/wp-calypso#96.

### CODE REVIEW

**Comment by bluefuton** [comment]
cc @aerych

File: client/reader/following-edit/subscribe-form.jsx [hunk 1]
```diff
@@ -98,11 +99,17 @@ var FollowingEditSubscribeForm = React.createClass( {
...
 	isWellFormedFeedUrl: function( parsedUrl ) {
-		if ( parsedUrl.hostname && parsedUrl.hostname.indexOf( '.' ) !== -1 ) {
-			return true;
+		if ( ! parsedUrl.hostname || parsedUrl.hostname.indexOf( '.' ) === -1 ) {
+			return false;
 		}
 
-		return false;
+		// Check for a valid-looking TLD
+		const lastHostnameSegment = last( parsedUrl.hostname.split( '.' ) );
```
Comment (blowery) [inline client/reader/following-edit/subscribe-form.jsx]: i think this is fine, but you could also use `lastIndexOf`, compare that to length - 2 (3?), and avoid the allocations that come with split.

Comment (bluefuton) [inline client/reader/following-edit/subscribe-form.jsx]: Thanks, I like that. Changing now.

**Comment by blowery** [comment]
:shipit:

**Pull request merged**